src/libsyntax/codemap.rs

```diff
@@ -592,8 +592,30 @@ impl CodeMap {
         }
     }
 
-    /// Given a `Span`, try to get a shorter span ending just after the first
-    /// occurrence of `char` `c`.
+    /// Given a `Span`, get a new `Span` covering the first token and all its trailing whitespace or
+    /// the original `Span`.
+    ///
+    /// If `sp` points to `"let mut x"`, then a span pointing at `"let "` will be returned.
+    pub fn span_until_non_whitespace(&self, sp: Span) -> Span {
```

Comment:
Nit: maybe we can implement this and `span_through_char` in terms of some common 

```rust
fn span_until_predicate(&self, predicate: impl Fn(char) -> bool) -> Span { ... }
```

?

Reply:
Oh, I see that the code is quite different. Never mind, not needed (yet).